File: auto-seq-editor.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import os
import sys
import re

input_file=sys.argv[1]
with open(input_file) as f:
    lines=f.readlines()

cc_file=os.getenv('CC_CLUSTER_INFO')
with open(cc_file) as f2:
    ccStr = f2.read()


with open(input_file,"w+") as f:
    for each in lines:
        if each.startswith("pick"):
            if ccStr.find(re.match(r'pick\s(\w+)\s(.*)',each).group(1))!=-1:
                each=each.replace("pick","squash")
                print(each+"")
        f.writelines(each)

File: auto-seq-editor.rb
#!/usr/bin/ruby

input_file = ARGV.shift
f.puts input_file
linesAll = IO.readlines(input_file)
lines=linesAll.select {|l| l.start_with?("pick") }
lines = lines.map {|l| id,
msg = l.scan(/^pick (\w+) (.*)/)[0] ;
                       { :line => l, :id => id, :msg => msg } }

cluster_info_file = ENV["CC_CLUSTER_INFO"] || "cc_cluster_info.txt"
cluster_targets = File.readlines(cluster_info_file).map {|l| l.chomp }

File.open(input_file, "w") do |f|
  linesAll[0..-1].each do |e|
    if (cluster_targets.find{|temp|/#{e[:id]}/=~temp})!=nil
      f.puts "squash #{e[:id]} #{e[:msg]}"
      puts "squash #{e[:id]} #{e[:msg]}"
    end
    else
       f.puts
  end
end